pennylane/ops/identity.py

```diff
@@ -451,10 +452,15 @@ def compute_decomposition(
     def eigvals(self):
         return self.compute_eigvals(self.data[0], n_wires=len(self.wires))
 
-    def matrix(self, wire_order=None):
-        n_wires = len(wire_order) if wire_order else len(self.wires)
+    def matrix(self, wire_order: Sequence = None):
+        n_wires = len(wire_order) if wire_order is None else len(self.wires)
```

Comment:
I think you meant this?

```suggestion
        n_wires = len(self.wires) if wire_order is None else len(wire_order)
```